src/04_video_container.js

```diff
@@ -323,6 +330,44 @@ class VideoContainerBase extends paella.DomNode {
 
 	get seekType() { return this._seekType; }
 
+	/**
+	* Synchronize the current time of the video container players
+	* @param {booloean} [repeat] default is False: to not set a timer to call syncVideos.
+	*                   set to True: to set the timer to call syncVideos again as last step.
+	*/
+	syncVideos(repeat) {
+		repeat = repeat || false;
+		let self = this;
+		paella.player.videoContainer.masterVideo().getVideoData().then(function (videoData) {
+			let currentTime = videoData.currentTime;
+			let streams = paella.player.videoContainer.streamProvider.videoPlayers;
+			let promises = [];
+			let updated = [];
+			let shorttime = 0.1;
+			streams.forEach(v => {
+				let diff = Math.abs(v.video.currentTime - currentTime);
+				base.log.debug(`About to test sync video=${v.video.id}, content=${v._stream.content}, paused=${v.video.paused}, timediff=${diff}`);
+				if ((v !== paella.player.videoContainer.streamProvider.mainAudioPlayer)
+					&& (!v.video.paused) && (diff > self._maxSyncDelay)) {
+					shorttime = (shorttime > currentTime ? shorttime: parseFloat(currentTime).toFixed(3));
+					base.log.debug(`Syncing video=${v.video.id}, content=${v._stream.content}, paused=${v.video.paused}, timediff=${diff}, settingToTime=${shorttime}`);
+					updated.push(v);
+					promises.push(v.setCurrentTime(shorttime));
+				}
+			});
+			Promise.all(promises).then(function() {
+				base.log.debug(`Synced ${updated.length} videos to ${shorttime}`);
+				if (repeat) {
+					setTimeout(function() {
+						paella.player.videoContainer.syncVideos(true);
+					}, self._videoSyncTimeMillis);
+				}
+			}).catch(error => {
+				base.log.debug(`Unable to synch video(s) to time '${shorttime}': ${error.message}`);
+			});
+		});
```

Comment:
Safari is not as seek friendly as other browsers. The Promises.all() stops setCurrentTime() promises from backing up on Safari.

The !v.video.paused prevents paused videos from having a seek done to them when the main audio player is still running. It reduces the amount of bandwidth needed when some of the videos are hidden and do not need to be in synch until they are enabled()